### DISCUSSION TITLE
Caret Annotation Width Adjustment

### DISCUSSION THREAD

**Discussion by aafreenrah** [pymupdf/PyMuPDF#4084] [Looking for help]
Caret annotation size is currently set as 20 x 20 rectangle size by default. no options to change it. In pdf, when we add it manually size is smaller. please give option to customize its size.

**Comment by JorjMcKie** [comment]
You can **_already_** change the size today by setting the annotation rectangle. This snippet adds 2 carets side by side, the second one has a size of 40 x 40:
```python
point = pymupdf.Point(100, 100)
annot = page.add_caret_annot(point)  # caret one

point = pymupdf.Point(120, 100)
annot = page.add_caret_annot(point)  # caret two
annot.set_rect(annot.rect + (0, 0, 20, 20))  # enlarge rect by 20 in each direction
```
Because this is so trivial to do, we have not added this as an extra functionality.

**Reply by aafreenrah** [reply]
Thank you for the reply. In this way, only position is getting changed. not the size. size remains the same!!

**Reply by JorjMcKie** [reply]
Don't understand. This is how the page looks like. What's wrong with the second caret?
![image](https://github.com/user-attachments/assets/06272522-8ae1-4049-a028-89bad72b4311)

**Comment by aafreenrah** [comment]
the resizing is working when I created pdf doc using "page.set_rotation(0)". It is not working in different document.
below is my code. resizing is not happening. 
   caret_position = fitz.Point(rects1[0].x1, rects1[0].y0)
   caret_annot = page.add_caret_annot(caret_position)
    caret_annot.set_rect(caret_annot.rect + (-4,-10,-12,-10))

**Reply by JorjMcKie** [reply]
I don't understand at all. Why are you referring to page rotation?

**Reply by aafreenrah** [reply]
Oh no. APologies. It is "page = doc.new_page()"

**Reply by JorjMcKie** [reply]
Hm, ok.
Your code shifts the caret's upper left corner 4 units to the left and 10 units up. The right border is moved 12 units to the left, effectively making the caret 8 units narrower, so the resulting width is 12.
The height remains the same (20) because both y values are reduced by the same amount (-10).
The resulting annot looks like this:
![image](https://github.com/user-attachments/assets/6b415cbd-b4d1-4aa2-a7b1-1374514beb07)

Instead of this:
![image](https://github.com/user-attachments/assets/502d6d1b-9fe1-4e28-81a3-f93055d6454a)

**Reply by aafreenrah** [reply]
<img width="155" alt="image" src="https://github.com/user-attachments/assets/26ca20e0-6a7e-416d-b070-64cd0ce42ccf">
In my case, the width is not changing. It should be narrower as you mentioned. mine is not empty document. is this issue document-specific? In empty document, resizing is working as expected for me.

**Reply by JorjMcKie** [reply]
There is no such dependency, it should always work.
